lib/spack/spack/cmd/checkbinaries.py

```diff
@@ -0,0 +1,247 @@
+    if match_list:
+        result = match_list
+    return result
+
+
+def get_mirror_rebuilds(mirror_name, mirror_url, pkg_change_set):
+    tty.msg('Checking for built specs on %s' % mirror_name)
+
+    # First fetch the index.html
+    index_path = os.path.join(mirror_url, _buildcache_path, 'index.html')
+    index_contents = None
+
+    try:
+        url = urlopen(index_path)
+    except URLError as urlErr:
+        tty.error('Unable to open url %s' % index_path)
+        tty.error(urlErr)
+        return None
+    except Exception as expn:
+        tty.error('Error getting index from mirror')
+        tty.error(expn)
+        return None
+
+    index_contents = url.read()
+
+    if not index_contents:
+        tty.error('Unable to retrieve index from mirror')
+        return None
+
+    yaml_list = parse_yaml_list(index_contents)
+
+    rebuild_list = []
+
+    for pkg_name in pkg_change_set:
+        tty.msg('yamls associated with %s' % pkg_name)
+
+        for pkg_yaml in yaml_list:
+            try:
+                pkg_yaml.index('-%s-' % pkg_name)
+                tty.msg('  %s' % pkg_yaml)
+            except Exception as expn:
+                continue
+
+            yaml_path = os.path.join(mirror_url, _buildcache_path, pkg_yaml)
+            yaml_contents = None
+
+            try:
+                yaml_url = urlopen(yaml_path)
+            except URLError as urlErr:
+                tty.error('Unable to open yaml url %s' % yaml_path)
+                tty.error('Mirror (%s: %s)' % (mirror_name, mirror_url))
+                tty.error(urlErr)
+                return None
+            except Exception as expn:
+                tty.error('Error getting yaml file: %s' % yaml_path)
+                tty.error('Mirror (%s: %s)' % (mirror_name, mirror_url))
+                tty.error(expn)
+                return None
+
+            yaml_contents = yaml_url.read()
+
+            if not yaml_contents:
+                tty.error('Error getting yaml contents: %s' % yaml_path)
+                tty.error('Mirror (%s: %s)' % (mirror_name, mirror_url))
+                return None
+
+            needs_rebuild = check_needs_rebuild(pkg_name, yaml_contents)
+
+            if needs_rebuild:
+                rebuild_list.append(pkg_yaml)
+
+    return rebuild_list
+
+
+def setup_parser(subparser):
+    subparser.add_argument(
+        '-b', '--base', action='store', default='develop',
+        help="select base branch for collecting list of modified files")
+
+    subparser.add_argument(
+        '-m', '--mirror_url', default=None, help='Additional mirror url')
+
+    # used to construct scope arguments below
+    scopes = spack.config.scopes()
+    scopes_metavar = spack.config.scopes_metavar
+
+    subparser.add_argument(
+        '--scope', choices=scopes, metavar=scopes_metavar,
+        default=spack.cmd.default_modify_scope(),
+        help="configuration scope containing mirrors to check")
+
+
+def checkbinaries(parser, args):
```

Comment:
`check-binaries` would be better - the function would be called `check_binares` in that case. does that work?

Reply:
Fine with me, I don't see why it wouldn't work.